src/create_links_sinks.py

```diff
@@ -0,0 +1,160 @@
+parser.add_argument('--regions', help="Double quoted, comma separated list of regions to create a sink and StackSet in. If not specified, a Sink/StackSet will be created in all default AWS regions", type=str, required=False)
+parser.add_argument('--excluded-accounts', help="[Optional] double quoted, comma separated list of accounts to exclude (ex. --exclude \"12345678910, 09876543210\"). Must exclude suspended accounts", type=str, required=False, default=None)
+
+# AWS Default regions
+aws_default_regions = ["us-east-2", "us-east-1", "us-west-1", "us-west-2",
+		"ap-south-1", "ap-northeast-3", "ap-northeast-2", "ap-southeast-1", "ap-southeast-2",
+		"ap-northeast-1", "ca-central-1", "eu-central-1", "eu-west-1", "eu-west-2", "eu-west-3",
+		"eu-north-1", "sa-east-1"]
+
+policy = """{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":"*","Action":["oam:CreateLink","oam:UpdateLink"],"Resource":"*","Condition":{"ForAllValues:StringEquals":{"oam:ResourceTypes":"AWS::CloudWatch::Metric"},"ForAnyValue:StringEquals":{"aws:PrincipalOrgID":"%s"}}}]}"""
+
+# Returns an instance of a client for a given service, in a given region, with a given profile
+def get_client(profile, service, region):
+    return boto3.Session(profile_name=profile, region_name=region).client(service)
+
+# Creates a sink in the provided region, and attaches the above policy to it
+def create_sink(region, profile, sink_name, organization):
+    # Get OAM client
+    client = get_client(profile, 'oam', region)
+
+    # Create an OAM sink in the specified region
+    try:
+        sink = client.create_sink(
+            Name=sink_name
+        )
+    except Exception:
+        raise Exception('creating sink')
+    
+    # Put organization in the policy string
+    formatted_policy = policy % organization
+
+    # Attach policy to sink
+    try:
+        client.put_sink_policy(
+            SinkIdentifier=sink['Arn'],
+            Policy=formatted_policy
+        )
+    except Exception:
+        raise Exception('adding policy to sink')
+    
+    return sink['Arn']
+
+# Renders link-cfn-template.yaml as a string with user provided arguments
+def render_template(monitoring_account_id, sink_arn):
+    env = Environment(
+        loader=FileSystemLoader('./'),
+        autoescape=select_autoescape())
+    template = env.get_template('link-cfn-template.yaml')
+    return template.render(MonitoringAccountID=monitoring_account_id, SinkARN=sink_arn)
+
+# Creates a stackset and stack instances that create links to the created sink
+def create_stackset(region, profile, sink_arn, organization_unit, stack_set_name, excluded_accounts):
+    # Get CFN Client
+    client = get_client(profile, 'cloudformation', region)
+
+    # Get account ID from caller identity
+    try:
+        account_id = get_client(profile, 'sts', region).get_caller_identity()['Account']
+    except Exception:
+        raise Exception('get account id')
```

Comment:
Maybe try this? We can preserve information from the original exception so that the error logs will be more informative about what went wrong.

```suggestion
    except Exception as e:
        raise Exception('get account id') from e
```

If that makes sense, we can roll it out to the exception handling elsewhere too

Reply:
rolled out